### ISSUE TITLE
[enhancement] regarding an install script

### ISSUE THREAD

**Issue by nichroot** [user18130814200115-2/riverbed#1]
hey! first of all i must say this is one of the coolest setups i've seen in a while, like the way it works is just genius. I really wanna try it out but couldnt really follow the steps you mentioned in the readme. Is there any way you might create an install script in the future? thanks!

**Comment by user18130814200115-2** [comment]
Thank you for the kind words.

An install script is definitely something I am hoping to implement. The issues with making it however are twofold.

Firstly, there is a lot of work left to be done before a script like this could be effectively implemented. I will track this work in this issue.

Secondly, I do not have the means to set up a good test-evnironment for such a script. I currently cannot easily setup fresh alpine installations, so I cannot fully test installing from scrach.

This might take some time. In the meantime, I have opened an issue #2 regarding the instalation instructions. Please go there to list what you found unclear and what you misunderstood.

**Comment by user18130814200115-2** [comment]
- [x] write install script
    + Install configuration files
    + Install apk wold
    + Compile and install custom packages
    + Somehow deal with non-apktools distributions
- [x] make luatile/riverbed-autogen.lua location agnotsic
- [ ] tag a stable release

**Comment by nichroot** [comment]
thanks for the reply! tbh i feel like its a me issue more than an issue with the instructions, i probably messed up stuff coz im currently using arch and tried to install it on arch as im in no way that expert in linux to use something like alpine. I'll try again and let yk if it goes right

**Comment by user18130814200115-2** [comment]
I see. For something like arch, you should be able to install the required packages with something like 

yay -S dbus-x11 font-jetbrains-mono-nerd river mkrundir waybar tofi fnott brightnessctl falkon capitaine-cursors grimshot consolekit2 wlr-randr pass-otp pipewire pipewire-pulse seatd-launch wbg wob wireplumber xdg-desktop-portal rtkit xdg-desktop-portal-lxqt zsh-syntax-highlighting cmd-polkit gtk-session-lock gtklock libinput-gestures river-luatile river-shifttags

Of course, Arch might use different package names for some of these programs, and some might have to come from the AUR.

I think, sinc arch uses systemd, you can launch river simply by calling `river` rather than `config/river/launch`.

Instructions for installing the configuration files remain the same.

**Comment by user18130814200115-2** [comment]
Also, an install script would not include arch support without some help from someone other than me. Essentially, if someone can tell me which package names are different in arch, then I would consider adding support at least for the install, possibly for the configuration as well.

**Comment by nichroot** [comment]
yeah there seems to be some problem when im trying on arch, but hold up im coming to the apk world (installing alpine rn) please wish me luck. also is there any particular reason you are using alpine? would love to know it

**Comment by user18130814200115-2** [comment]
I use apline for two reasons. Firstly, apk-tools is the best package manager I have ever used. It is easy to keep track of packages with /etc/apk/world, and there are good tools avaible for building packages localy (namely abuild-rootbld which builds a package in a clean environment). Plus, apk is the fastest package manager available, easily twice as fast as apt, and maybe 1.5 times faster than pacman.

Generally though, there just seems to a lot exciting stuff happening in the Alpine world, new init systems, updates to the package manager, super super robjust tooling, and a focus on stability while bringing new features fast. I also like that apline has both a statbe branch and a rolling edge.

**Comment by nichroot** [comment]
> I use apline for two reasons. Firstly, apk-tools is the best package manager I have ever used. It is easy to keep track of packages with /etc/apk/world, and there are good tools avaible for building packages localy (namely abuild-rootbld which builds a package in a clean environment). Plus, apk is the fastest package manager available, easily twice as fast as apt, and maybe 1.5 times faster than pacman.
> 
> Generally though, there just seems to a lot exciting stuff happening in the Alpine world, new init systems, updates to the package manager, super super robjust tooling, and a focus on stability while bringing new features fast. I also like that apline has both a statbe branch and a rolling edge.

got it! also do you think i'll be good to copy your setup if i follow this video for installing alpine and river?

https://www.youtube.com/watch?v=YAUUT4ey7eU

**Comment by user18130814200115-2** [comment]
Yes that video is fine. The only major difference is that you should not run
rc-service seatd start, instead, install seatd-launch and use the config/river/launch script.

Minor points is that my setup uses Doas rather than sudo, but this is not problematic in any way.
Same goes for using ash instead of Bash.

**Comment by user18130814200115-2** [comment]
I just pushed an experimental install script. I have not extensively tested it though...

**Comment by nichroot** [comment]
should i run it after installing river or does the script install river and other packages as well?

**Comment by user18130814200115-2** [comment]
The should run from a fresh alpine install without river present. All packages will be installed and compiled for you.

**Comment by user18130814200115-2** [comment]
One thing which must still be fixed is the loading done by river-luatile.

For now, after installing. open `~/.config/river-luatile/layout.lua` and change `dofile('/home/user/.config/river-luatile/riverbed-autogen.lua')` to contain your username. For example, if your username is john, change the line to `dofile('/home/john/.config/river-luatile/riverbed-autogen.lua')

EDIT: This is fixed with the latest push
`

**Comment by nichroot** [comment]
i'll do it and let you know how it goes

**Comment by nichroot** [comment]
i'll list the errors i got there #2

**Comment by user18130814200115-2** [comment]
Experimental support was added for nixos installation which should be more reliable due to its declarative nature. Currently, it is alpha of course, and only works on NixOs itself, though ti should be portable to nix with home-manager on other distributions.

**Comment by nichroot** [comment]
hey! so will it work fine on an install from the nixOS graphical installer?

**Comment by user18130814200115-2** [comment]
Yes! In fact, I used the graphical installer to install nixos with gnome and from there crafted the riverbed.nix file. After installing, just include the riverbed.nix file in your configuration.nix and rebuild. let me know if you encounter any problems :)

**Comment by nichroot** [comment]
Awesome, thank you so much! will try it out and lyk
